### ISSUE TITLE
Allow a machine to configure the default Python interpreter version

### ISSUE THREAD

**Issue by lostintangent** [microsoft/vscode-jupyter#9997]
Codespaces recently updated its base image to pre-install Jupyter along with a handful of ubiquitous ML libraries (e.g. `pandas`, `torch`, etc.). These libraries are installed into the context of the latest version of Python that's installed (currently 3.10.4), which is also set as the default Python version in the user's `$PATH`. Because of this, it would be ideal if when a user opened an `*.ipynb` file in a Codespace, that we could have it automatically select the kernel for the latest Python version. That way, there's a seamless continuity between the terminal and notebooks, without the user needing to select a kernel manually, and potentially get confused.

For example, the following is a screenshot of what the kernel picker looks like in a Codespace, and it would be awesome if we could somehow configure the environment to pre-select `Python 3.10.4 64-bit`.

![image](https://user-images.githubusercontent.com/116461/167996426-119f3eaf-0cf3-4b95-a4bf-a957195057a1.png)

**Comment by lostintangent** [comment]
Interestingly enough, if I run the `Create: New Jupyter Notebook` command, the new untitled notebook seems to automatically select the desired kernel version. But as soon as you save the untitled notebook, it reloads, and the kernel picker switches back to `Select kernel`.

So if would be cool if there was a way to automatically select a preferred Python version for `*.ipynb` files, in the same way that appears to occur for untitled files.

**Comment by IanMatthewHuff** [comment]
Yeah, this suggestions ties in well with a lot of work that we are currently doing around selecting a preferred kernel.

https://github.com/microsoft/vscode-jupyter/issues/9615

Currently when we open up a new blank notebook file without any existing metadata if the language is Python then we actually delegate to using the current active Python interpreter from the Python extension for our preferred kernel. I would actually expect that in this case (without tweaking any python extension settings) that the $PATH setting would be honored by the python extension and the correct kernel would be selected. I think that is what you are seeing the `Create: New Jupyter Notebook` case. If you are getting something different when opening a .ipynb I believe that there might be something in that file's metadata that's causing the Jupyter extension to prefer a different kernel from the ones available.

Just to mention, going forward one of our considerations might be to allow configuring a default kernel on the user or workspace level and have that win out over any heuristic guesses or python extensions settings. Might be an option for better supporting Codespaces.

**Comment by IanMatthewHuff** [comment]
Is this change live in Codespaces? I'm not a regular code spaces user myself, but I'd love to take a peek to see what the default behavior looks like here.

One other side note, I'm not sure how the lifetime of codespaces are handled, but in general installing packages and using the main python interpreter directly as a kernel is something that we are looking at moving away from. We have lots of issues with global python installs getting polluted with module installs and getting folks into bad situations. What might be a better solutions here is to instead install a separate jupyter kernel with DS packages installed independent of the main python interpreter.

**Comment by IanMatthewHuff** [comment]
Yeah, I'd defer to Brett on that. Given the statement about ephemerality of codespaces working with the global interpreter seems fine.

This bit here does sound like at least a bit of a bug:
> But as soon as you save the untitled notebook, it reloads, and the kernel picker switches back to Select kernel.

There is a known issue with the core notebook that when you save it, it loses the current controller selection. However when it is saved it should then run the same algorithm as when first created. If it correctly selects the right kernel when created it should also reselect that kernel when saved. I'll take a peek at that.

**Comment by DonJayamanne** [comment]
Transferring to Python extension, as this should ideally be fixed in the Python extension.
I.e. if Python extension were to have the `Active Interpreter` of the workspace point to the item thats been suggested, then the Jupter extension would pick that up
